site/app/libraries/database/PostgresqlDatabaseQueries.php

```diff
@@ -939,17 +939,67 @@ public function getTeamsById(array $team_ids) {
         return $teams;
     }
 
+    /**
+     * Maps sort keys to an array of expressions to sort by in place of the key.
+     *  Useful for ambiguous keys or for key alias's
+     */
+    const graded_gradeable_key_map = [
+        'registration_section' => [
+            'SUBSTRING(u.registration_section, \'^[^0-9]*\')',
+            'COALESCE(SUBSTRING(u.registration_section, \'[0-9]+\')::INT, -1)',
+            'SUBSTRING(u.registration_section, \'[^0-9]*$\')',
+            'SUBSTRING(team.registration_section, \'^[^0-9]*\')',
+            'COALESCE(SUBSTRING(team.registration_section, \'[0-9]+\')::INT, -1)',
+            'SUBSTRING(team.registration_section, \'[^0-9]*$\')'
+        ],
+        'rotating_section' => [
+            'u.rotating_section',
+            'team.rotating_section'
+        ]
+    ];
+
+    /**
+     * Generates the ORDER BY clause with the provided sorting keys
+     * @param string[] $sort_keys
+     * @param array $key_map A map from sort keys to arrays of expressions to sort by instead
+     *          (see self::graded_gradeable_key_map for example)
+     * @return string
+     */
+    private static function generateOrderByClause(array $sort_keys, array $key_map) {
+        if ($sort_keys !== null) {
+            if (!is_array($sort_keys)) {
+                $sort_keys = [$sort_keys];
+            }
+            // Use simplified expression for empty keymap
+            if (empty($key_map)) {
+                return 'ORDER BY ' . implode(',', $sort_keys);
+            }
+            return 'ORDER BY ' . implode(',', array_map(function ($key_ext) use ($key_map) {
+                    $split_key = explode(' ', $key_ext);
+                    $key = $split_key[0];
+                    $order = '';
+                    if (count($split_key) > 1) {
+                        $order = $split_key[1];
+                    }
```

Comment:
Otherwise specify default order? Or is ORDER BY with no order still allowed?

Reply:
`ORDER BY` defaults to `ASC` if none specified and I think we should preserve that behavior

Reply:
👍